src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,210 @@
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import com.revrobotics.RelativeEncoder;
+import com.spikes2212.command.drivetrains.smartmotorcontrollerdrivetrain.SparkMaxTankDrivetrain;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.control.TrapezoidProfileSettings;
+import com.spikes2212.dashboard.Namespace;
+import edu.wpi.first.math.controller.RamseteController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc.robot.RobotMap;
+
+import java.util.function.Supplier;
+
+public class Drivetrain extends SparkMaxTankDrivetrain {
+
+    public static final double WHEEL_DIAMETER_IN_INCHES = 6;
+    public static final double GEAR_RATIO = 1 / 12.755;
+    public static final double INCHES_TO_METERS = 0.0254;
+    public static final double DISTANCE_PER_PULSE = WHEEL_DIAMETER_IN_INCHES * GEAR_RATIO * Math.PI * INCHES_TO_METERS;
+
+    public static final double TRACK_WIDTH = 0.57;
+
+    public static final int SECONDS_IN_MINUTE = 60;
+
+    private static Drivetrain instance;
+
+    private final AHRS gyro;
+
+    private final DifferentialDriveOdometry odometry;
+    private final DifferentialDriveKinematics kinematics;
+    private final RamseteController ramseteController;
+    private final Field2d field2d;
+
+    private final RelativeEncoder leftEncoder;
+    private final RelativeEncoder rightEncoder;
+
+    private final Namespace leftPIDNamespace = namespace.addChild("left pid");
+    private final Supplier<Double> kPLeft = leftPIDNamespace.addConstantDouble("kP", 0);
+    private final Supplier<Double> kILeft = leftPIDNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kDLeft = leftPIDNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> waitTimeLeft = leftPIDNamespace.addConstantDouble("wait time", 0);
+    private final Supplier<Double> toleranceLeft = leftPIDNamespace.addConstantDouble("tolerance", 0);
+    private final PIDSettings leftPIDSettings;
+
+    private final Namespace rightPIDNamespace = namespace.addChild("right pid");
+    private final Supplier<Double> kPRight = rightPIDNamespace.addConstantDouble("kP", 0);
+    private final Supplier<Double> kIRight = rightPIDNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kDRight = rightPIDNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> waitTimeRight = rightPIDNamespace.addConstantDouble("wait time", 0);
+    private final Supplier<Double> toleranceRight = rightPIDNamespace.addConstantDouble("tolerance", 0);
+    private final PIDSettings rightPIDSettings;
+
+    private final Namespace anglePIDNamespace = namespace.addChild("angle pid");
```

Comment:
what does angle mean? could I have multiple pid loops that check the robot's angle?